packages/core/src/browser/external-uri-service.ts

```diff
@@ -31,34 +37,43 @@ export class ExternalUriService {
      * Use `parseLocalhost` to retrieve localhost address and port information.
      */
     resolve(uri: URI): MaybePromise<URI> {
-        const localhost = this.parseLocalhost(uri);
-        if (localhost) {
-            return this.toRemoteUrl(uri, localhost);
+        const address = this.parseLocalhost(uri);
+        if (address) {
+            return this.toRemoteUrl(uri, address);
         }
         return uri;
     }
 
-    protected toRemoteUrl(uri: URI, localhost: { address: string, port: number }): URI {
-        const host = this.toRemoteHost(localhost);
-        return new Endpoint({ host }).getRestUrl().withPath(uri.path).withFragment(uri.fragment).withQuery(uri.query);
-    }
-
-    protected toRemoteHost(localhost: { address: string, port: number }): string {
-        return `${window.location.hostname}:${localhost.port}`;
-    }
-
-    parseLocalhost(uri: URI): { address: string, port: number } | undefined {
+    parseLocalhost(uri: URI): AddressPort | undefined {
         if (uri.scheme !== 'http' && uri.scheme !== 'https') {
-            return undefined;
+            return;
         }
         const localhostMatch = /^(localhost|127\.0\.0\.1|0\.0\.0\.0):(\d+)$/.exec(uri.authority);
         if (!localhostMatch) {
-            return undefined;
+            return;
         }
         return {
             address: localhostMatch[1],
             port: +localhostMatch[2],
         };
     }
 
+    protected toRemoteUrl(uri: URI, address: AddressPort): URI {
+        return new Endpoint({ host: this.toRemoteHost(address) })
+            .getRestUrl()
+            .withPath(uri.path)
+            .withFragment(uri.fragment)
+            .withQuery(uri.query);
+    }
+
+    protected toRemoteHost(address: AddressPort): string {
+        return `${this.getRemoteHost()}:${address.port}`;
```

Comment:
Will there always be a port here?

Reply:
Given how `AddressPort` works/is defined: yes.

Reply:
Yes, I suppose there likely will be, since this is only called for `localhost`.